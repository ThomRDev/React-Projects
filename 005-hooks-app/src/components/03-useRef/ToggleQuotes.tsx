import { useState } from "react"
import QuotesApp from "../02-useEffect/Quotes"

const ToggleQuotes = () => {
    const [show,setShow] = useState(true)
    return <div>
        <h1>ToggleQuotes</h1>
        <button className="btn btn-secondary"
            onClick={()=>setShow(!show)}
        >{show ? "Ocultar" :"Mostrar"}</button>
        {
            show && <QuotesApp />
        }
    </div>
}
export default ToggleQuotes