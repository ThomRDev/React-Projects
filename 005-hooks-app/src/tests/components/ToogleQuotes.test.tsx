import renderer from "react-test-renderer"
import ToggleQuotes from "../../components/03-useRef/ToggleQuotes"

describe('Test ToogleQuotes', () => {
  
  // enzime u.u
  // const wrapper = shallow(<ToggleQuotes />)

  test('snapshot', () => {
    const tree = renderer.create(<ToggleQuotes />).toJSON()
    expect(tree).toMatchSnapshot()

    // expect(wrapper).toMatchSnapshot()
    // expect(wrapper.find("QuotesApp").exists()).toBe(true) //por defecto es true
  })

  test('ToggleQuotes hidden', () => {

    // eventWrapper.find("button").simulate("click")
    // expect(wrapper.find("QuotesApp").exists()).toBe(false)
  })
})