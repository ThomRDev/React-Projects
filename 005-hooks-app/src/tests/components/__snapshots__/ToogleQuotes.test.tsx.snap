// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ToogleQuotes snapshot 1`] = `
<div>
  <h1>
    ToggleQuotes
  </h1>
  <button
    className="btn btn-secondary"
    onClick={[Function]}
  >
    Ocultar
  </button>
  <div>
    <div
      className="alert alert-info text-center"
    >
      loading ... 
    </div>
    <button
      className="btn btn-primary disabled"
      disabled={true}
      onClick={[Function]}
    >
      Cambiar
    </button>
  </div>
</div>
`;
